app/controllers/admin/deduplicate_patients_controller.rb

```diff
@@ -3,15 +3,19 @@ class Admin::DeduplicatePatientsController < AdminController
   DUPLICATE_LIMIT = 250
 
   def show
-    authorize { current_admin.accessible_organizations(:manage).any? }
+    authorize { current_admin.accessible_facilities(:manage).any? }
 
-    duplicate_patient_ids = PatientDeduplication::Strategies.identifier_excluding_full_name_match(limit: DUPLICATE_LIMIT)
+    duplicate_patient_ids =
+      PatientDeduplication::Strategies.identifier_excluding_full_name_match(
+        limit: DUPLICATE_LIMIT,
+        facilities: current_admin.accessible_facilities(:manage)
+      )
     @duplicate_count = duplicate_patient_ids.count
     @patients = Patient.where(id: duplicate_patient_ids.sample).order(recorded_at: :asc)
   end
 
   def merge
-    authorize { current_admin.accessible_organizations(:manage).any? }
+    authorize { current_admin.accessible_facilities(:manage).any? }
```

Comment:
In addition to this check, we should validate that the `duplicate_patient` belong to authorized facilities as well.